src/dict.c

```diff
@@ -1433,19 +1420,14 @@ static int dictTypeResizeAllowed(dict *d, size_t size) {
                     (double)d->ht_used[0] / DICTHT_SIZE(d->ht_size_exp[0]));
 }
 
-/* Expand the hash table if needed */
-static void _dictExpandIfNeeded(dict *d)
-{
-    /* Automatic resizing is disallowed. Return */
-    if (d->pauseAutoResize > 0) return;
-
+int dictExpandIfNeeded(dict *d) {
```

Comment:
I'd like to have comments here and around `dictShrinkIfNeeded` to explain the meaning of the return values: "DICT_OK" indicates a successful expand or the current dictionary is undergoing resize, while "DICT_ERR" indicates that resize has not been triggered.